mutation/mutation_partition.cc

```diff
@@ -478,6 +484,22 @@ mutation_partition::apply(const schema& s, mutation_partition_view p,
     apply_monotonically(s, std::move(p2), no_cache_tracker, app_stats, is_preemptible::no, res);
 }
 
+future<>
+mutation_partition::apply_gently(const schema& s, mutation_partition_view p,
+        const schema& p_schema, mutation_application_stats& app_stats) {
+    // FIXME: Optimize
+    mutation_partition p2(*this, copy_comparators_only{});
+    partition_builder b(p_schema, p2);
+    co_await p.accept_gently(p_schema, b);
+    if (s.version() != p_schema.version()) {
+        p2.upgrade(p_schema, s);
+    }
+    apply_resume res;
+    while (apply_monotonically(s, std::move(p2), no_cache_tracker, app_stats, is_preemptible::yes, res) == stop_iteration::no) {
+        co_await yield();
+    }
```

Comment:
There's a good chance that this regresses writes.

We may want to fast-path small mutations.

Also, it can break atomicity for memtable/cache updates. While we yield, we see a partial result.

A solution can be to lock the entire cache/memtable, but it's hairy.

Reply:
It's not used on the hot path.
I added a comment in a later version noting that it's not suitable when apply must be atomic.